std-bits/tableau/src/main/resources/META-INF/native-image/org/enso/tableau/resource-config.json

```diff
@@ -0,0 +1,22 @@
+{
+    "resources": {
+        "includes": [
+            {
+                "pattern": "\\QMETA-INF/services/com.tableau.hyperapi.impl.SharedLibraryProvider\\E"
+            },
+            {
+                "pattern": "\\Qlinux-x86-64/libtableauhyperapi.so\\E"
```

Comment:
These libraries should be extracted into `polyglot/lib` directory.

Reply:
At one time there is only a single OS-specific jar with the library. `resource-config` is simply not aware of OS/Arch so listing them here is of little harm, I believe. 

I haven't changed in any way the way we package/include shared library and os/arch-specific jar has always been included in `polyglot`. What do we gain by this and why this seems to be NI-specific to you and should be addressed in this PR?

Reply:
- @Akirathan worked on #11874 
- making the `polyglot/lib` an easy to use _standard way of packaging JNI libraries_
    - makes the distribution smaller: #11483
    - makes the startup faster (no extraction of a resource on start)
    - avoids duplicating the library between JAR and _native executable_
- the whole system was successfully used in #12021
- we should use it for all our JNI libraries (tableau, postgres, sqlite)
- unless there is a reason not to

Reply:
Yeap, totally makes sense, I somehow forgot about it. Thanks for bring it up.

Reply:
This was a bit more messy than initially expected. Had to provide a custom classloader logic that intercepts hyper api search that is done in Tableau's library.